tests/python/rest_api/test_tasks.py

```diff
@@ -674,6 +685,77 @@ def test_create_task_with_cloud_storage_files(
             self._USERNAME, task_spec, data_spec, content_type="application/json", org=org
         )
 
+    @pytest.mark.parametrize(
+        "cloud_storage_id, manifest, pattern, sub_dir, task_size, org",
+        [
+            (1, "manifest.jsonl", "*", True, 3, ""),  # public bucket
+            (1, "manifest.jsonl", "test/*", True, 3, ""),
+            (1, "manifest.jsonl", "test/sub*1.jpeg", True, 1, ""),
+            (1, "manifest.jsonl", "*image*.jpeg", True, 3, ""),
+            (1, "manifest.jsonl", "wrong_pattern", True, 0, ""),
+            (1, "abc_manifest.jsonl", "[a-c]*.jpeg", False, 2, ""),
+            (1, "abc_manifest.jsonl", "[d]*.jpeg", False, 1, ""),
+            (1, "abc_manifest.jsonl", "[e-z]*.jpeg", False, 0, ""),
+        ],
+    )
+    def test_create_task_with_file_pattern(
+        self, cloud_storage_id, manifest, pattern, sub_dir, task_size, org, cloud_storages
+    ):
+        # prepare dataset on the bucket
+        prefixes = ("test_image_",) * 3 if sub_dir else ("a_", "b_", "d_")
+        images = generate_image_files(3, prefixes=prefixes)
+        s3_client = make_client()
+
+        cloud_storage = cloud_storages[cloud_storage_id]
+
+        for image in images:
+            s3_client.create_file(
+                data=image,
+                bucket=cloud_storage["resource"],
+                filename=f"{'test/sub/' if sub_dir else ''}{image.name}",
+            )
+
+        with NamedTemporaryFile(mode="wb+", suffix="manifest.jsonl") as tmp_file:
+            tmp_name = tmp_file.name
+            tmp_manifest = ImageManifestManager(manifest_path=tmp_name)
+            tmp_manifest.link(sources=images)
+            tmp_manifest.create()
+            tmp_file.seek(0)
+            s3_client.create_file(
+                data=tmp_file.read(),
+                bucket=cloud_storage["resource"],
+                filename=f"test/sub/{manifest}" if sub_dir else manifest,
+            )
+
+        task_spec = {
+            "name": f"Task with files from cloud storage {cloud_storage_id}",
+            "labels": [
+                {
+                    "name": "car",
+                }
+            ],
+        }
+
+        data_spec = {
+            "image_quality": 75,
+            "use_cache": True,
+            "cloud_storage_id": cloud_storage_id,
+            "server_files": [f"test/sub/{manifest}" if sub_dir else manifest],
+            "pattern": pattern,
+        }
+
+        if task_size:
+            task_id = self._test_create_task(
+                self._USERNAME, task_spec, data_spec, content_type="application/json", org=org
+            )
+
+            with make_api_client(self._USERNAME) as api_client:
+                (task, response) = api_client.tasks_api.retrieve(task_id, org=org)
+                assert response.status == HTTPStatus.OK
+                assert task.size == task_size
+        else:
+            self._test_cannot_create_task(self._USERNAME, task_spec, data_spec)
```

Comment:
It would be better to also check the failure reason.

Reply:
Fixed